### PR TITLE
Feature/play 2.4.4

### PR DESCRIPTION
**Pull request by lloydmeta**
This PR aims to allow us to upgrade to Play 2.4.2. As of writing, all tests pass and a manual smoke test works as well

It is marked WIP because it should not be merged yet due to:

- Broken Swagger (not 2.4 compatible) [keep an eye here](https://github.com/swagger-api/swagger-core/issues/918)
- Broken Kenshoo metrics (not 2.4 compatible)
- Plugin system being deprecated

Odd things:
- Calling #reloadCache in the CachingRepository initialiser causes `exceptionInInitializerError: null` errors when trying to resolve Skinny ORM ref relations.

### CODE REVIEW

**Comment by lloydmeta** [comment]
Just noticed this fails on Travis but works when testing locally. Odd..

**Comment by lloydmeta** [comment]
Removing WIP tag because this is now ready.

All aforementioned issues have been resolved:

- Broken Swagger (not 2.4 compatible) **Fixed**
- Broken Kenshoo metrics (not 2.4 compatible) **Fixed**
- Plugin system being deprecated **Ignorable** (the API is deprecated but still usable)

Odd things:
- Calling #reloadCache in the CachingRepository initialiser causes `exceptionInInitializerError: null` errors when trying to resolve Skinny ORM ref relations. **Fixed**

File: scala-ws-client/src/main/scala/com/m3/octoparts/ws/OctoClient.scala [hunk 1]
```diff
@@ -29,8 +29,11 @@ import com.m3.octoparts.json.format.ConfigModel._ // For serdes of the models
  */
 class OctoClient(val baseUrl: String, protected val clientTimeout: FiniteDuration, protected val extraWait: FiniteDuration = 50.milliseconds)(implicit val octoPlayApp: Application) extends OctoClientLike {
 
-  protected def wsHolderFor(url: String, timeout: FiniteDuration) =
-    WS.url(url).withRequestTimeout((timeout + extraWait).toMillis.toInt)
+  private val client = WS.client
```
Comment (lloydmeta) [inline scala-ws-client/src/main/scala/com/m3/octoparts/ws/OctoClient.scala]: TODO: Change this to be passed in via injection.

File: vaygrant/Vagrantfile [hunk 2]
```diff
@@ -14,7 +14,7 @@ Vagrant.configure(VAGRANTFILE_API_VERSION) do |config|
 
   # Create a forwarded port mapping which allows access to a specific port
   # within the machine from a port on the host machine.
-  config.vm.network :forwarded_port, guest: 9000, host: 9000 # For accessing octoparts
+#  config.vm.network :forwarded_port, guest: 9000, host: 9000 # For accessing octoparts
```
Comment (xevix) [inline vaygrant/Vagrantfile]: Is this no longer necessary?

File: test/controllers/AdminControllerSpec.scala [hunk 3]
```diff
@@ -140,7 +140,7 @@ class AdminControllerSpec extends FunSpec
 
   it("should show a list of parts") {
     val repository = mock[MutableConfigsRepository]
-    val adminController = new AdminController(cacheOps = DummyCacheOps, repository = repository)
+    val adminController = new AdminController(cacheOps = DummyCacheOps, repository = repository, messagesApi = messagesApi)
```
Comment (xevix) [inline test/controllers/AdminControllerSpec.scala]: Ah, so much i18n, haha.

File: conf/application.conf [hunk 4]
```diff
@@ -49,9 +47,8 @@ db {
     password=${?OCTOPARTS_DB_PASSWORD}
 
     # Connection pool settings
-    poolInitialSize=1
-    poolMaxSize=4
-    poolFactoryName = "commons-dbcp2"
+    poolInitialSize=10
```
Comment (xevix) [inline conf/application.conf]: :+1:

File: app/com/m3/octoparts/wiring/assembling/SwaggerScanSupport.scala [hunk 5]
```diff
@@ -0,0 +1,103 @@
...
+   */
+  protected def initSwagger(components: ApplicationComponents): Unit = {
+    val scanner = new SwaggerScanner(components.router, components.application, components.applicationLifecycle)
+    scanner.scan()
+  }
+
+}
+
+/**
+ * Ghetto version of [[pl.matisoft.swagger.SwaggerPluginProvider]] without dependence on
```
Comment (xevix) [inline app/com/m3/octoparts/wiring/assembling/SwaggerScanSupport.scala]: :laughing:

File: vaygrant/Vagrantfile (hunk 2 above)
Comment (lloydmeta) [inline vaygrant/Vagrantfile]: No it was getting in the way on my machine. When I had Vagrant running (for the DB and Memcached, etc), my Play on my host machine wasn't able to bind to 9000 :(

I should revert this.

**Comment by xevix** [comment]
Not sure if related, but while building Vagrant box from scratch.

```
TASK: [td_agent | Check for fluent-plugin-elasticsearch gem] ****************** 
failed: [default] => {"changed": true, "cmd": "/usr/lib64/fluent/ruby/bin/fluent-gem list fluent-plugin-elasticsearch -i", "delta": "0:00:00.117966", "end": "2015-12-06 09:53:54.716964", "rc": 1, "start": "2015-12-06 09:53:54.598998", "warnings": []}
stdout: false
...ignoring
```

**Comment by xevix** [comment]
```
[warn] /octoparts/plugins/auth-plugin-api/src/main/scala/com/m3/octoparts/auth/OctopartsAuthPlugin.scala:35: trait Plugin in package api is deprecated: Use modules instead. See `play.api.inject.Module`
[warn] trait OctopartsAuthPlugin extends Plugin {
[warn]                                   ^
[warn] one warning found
```

**Comment by xevix** [comment]
Also not sure if related, but it seems zipkin-web can't find its templates properly. If this is a separate issue, I'll open another issue.

`sudo less /var/log/supervisor/zipkin-web.log`
```
INF [20151206-10:07:45.574] finagle: Finagle version 6.5.2 (rev=69027d3638f46795ac7f8cd9f44defc795c5d080) built at 20130723-105853
INF [20151206-10:07:46.123] web: Finatra service started in port: 8085
INF [20151206-10:50:50.318] finatra: GET /
ERR [20151206-10:50:51.920] finatra: Internal Server Error
ERR [20151206-10:50:51.920] finatra: java.io.FileNotFoundException: /opt/zipkin/zipkin-web/zipkin-web/src/main/resources/templates/layouts/application.mustache (No such file or directory)
```

**Comment by xevix** [comment]
Also seems that monitoring links are missing (no <a> tags). Again, not sure if related.

octoparts_missing_links

**Comment by xevix** [comment]
Other than the above issues which may be unrelated, quick smoketest and unit tests seem to be flying. Awesome work!

**Comment by lloydmeta** [comment]
> Not sure if related, but while building Vagrant box from scratch.

This one is fine. It's a check for the existence of a gem, and if it fails, we set a variable value, but we don't want the failure to stop the provisioning process (it is an expected failure).

> [Reference to Plugin deprecation]

Yeah, Plugins are deprecated but they should still work. We should file an issue for upgrading to Module for our AuthPlugin interface after this is merged.

> Also not sure if related, but it seems zipkin-web can't find its templates properly. If this is a separate issue, I'll open another issue.

Oddly, I just did a full provision from scratch and my log doesn't have that error. I wonder if my changing of the CentOS box in order to reproduce your issue is related...

> Also seems that monitoring links are missing (no tags). Again, not sure if related.

This is unrelated. The default configuration does not have these links set (they are dependent on end-user infra)

**Comment by xevix** [comment]
@lloydmeta 

> Oddly, I just did a full provision from scratch and my log doesn't have that error. I wonder if my changing of the CentOS box in order to reproduce your issue is related...

I think you have to try browsing to it first. Have the box running, and then go to http://localhost:8085/ and it should fail.

**Comment by lloydmeta** [comment]
> I think you have to try browsing to it first. Have the box running, and then go to http://localhost:8085/ and it should fail.

Ah ok. Was able to reproduce it and fixed it in 2cfd38c

**Comment by xevix** [comment]
Ok, I can confirm that Zipkin is fixed, and traces are properly coming in. Nice fix!

That wraps it up for the review. Next up is running this thing through its paces in a real system.

File: project/Dependencies.scala [hunk 6]
```diff
@@ -157,12 +157,15 @@ object Dependencies {
     scalatestPlay,
     scalacheck,
     groovy,
-    scalikeJdbcTest
+    scalikeJdbcTest,
```
Comment (lloydmeta) [inline project/Dependencies.scala]: Hmm, can't remember why I added these here.

File: vaygrant/Vagrantfile (hunk 2 above)
Comment (xevix) [inline vaygrant/Vagrantfile]: @lloydmeta Ah almost forgot, can you revert this?

**Comment by lloydmeta** [comment]
Needs #181

**Comment by xevix** [comment]
@lloydmeta Since plugin support has been ripped out, would this then close #181 or is there something else?

**Comment by lloydmeta** [comment]
@xevix You're right this would close #181. Let's remember to do that.

**Comment by lloydmeta** [comment]
Note:

- `Streamer` class still uses injected ActorSystem
- Do a search for ^ and purge

**Comment by lloydmeta** [comment]
@xevix I've had this deployed internally for a good few hours now and everything seems operational (AuthHandler implementation, Octoparts itself, Zipkin, Hystrix+Turbine integration). If this looks good to you, please click the merge button and we can publish release versions of the dependencies afterwards.

**Comment by xevix** [comment]
Nice. As we come to a close I'm looking at services exposed in the Vagrantfile and I don't recognize or know that we've tested all of them, so I'd like to confirm that they're all fine.

- [X] config.vm.network :forwarded_port, guest: 9000, host: 9000 # For accessing octoparts
- [X] config.vm.network :forwarded_port, guest: 8080, host: 8080 # For accessing hystrix-dashboard from host
- [X] config.vm.network :forwarded_port, guest: 8085, host: 8085 # For accessing zipkin-web from host
- [ ] config.vm.network :forwarded_port, guest: 8443, host: 8443
- [ ] config.vm.network :forwarded_port, guest: 5432, host: 5432 # PSQL
- [X] config.vm.network :forwarded_port, guest: 11211, host: 11211 # Memcached
- [ ] config.vm.network :forwarded_port, guest: 5005, host: 5005 # for debugging
- [ ] config.vm.network :forwarded_port, guest: 9200, host: 9200 # Elasticsearch
- [ ] config.vm.network :forwarded_port, guest: 80, host: 5080 # Kibana @ nginx

In particular, I know 8443 is probably for some SSL server, but which? And 5005, debugging what in particular?

I've marked the boxes I know and leave the rest to you.

**Comment by lloydmeta** [comment]
I've ticked off PSQL, debugging (typical port for JVM debugger) and 8443 (Hystrix tries to listen on this port).

**Comment by lloydmeta** [comment]
Just verified that ElasticSearch + Kibana are working as well so ticked those too.

Also double checked via Zipkin that our Part-level caching is still working, which is A Good Thing.

**Comment by xevix** [comment]
LGTM. Awesome work.

**Comment by xevix** [comment]
:+1:

**Comment by lloydmeta** [comment]
:+1:

**Pull request merged**